Implement plotting functionality

busykoala opened the pull request:
Matplotlib has a great interface working together with pandas.
Therefore I chose to use this package for plotting the graphs.

Because we are still using Python 2.7 it was quite a pain to find versions that compile and/or work together.

Matplotlib is using the `Agg` backend because the CI does not support tkinter.

The graphs are plotted and then converted into base64. This allows rendering them within an image tag easily.

mbaechtold commented on ftw/linkchecker/tests/helpers.py:
```diff
@@ -47,6 +55,37 @@ def generate_test_data_excel_workbook():
+    file_.setFilename(ASSET_NAME)
+    transaction.commit()
+
+    if upload_two:
```
This is just a personal advice (you don't have to change it now): I tend to put refactoring (like moving this function to another module) in a separate commit, before adding new functionality. This makes diffing between the commits much easier, which is useful for debugging.

busykoala replied:
I added a separate commit for moving the test function. Thanks for the advice. It makes totally sense to me.

mbaechtold reviewed (changes requested):
Could you please split the first commit into 3 commits:
- Sorting the dependencies
- Moving of production dependencies
- Addition of new dependencies

Commit:
Add setup for start plotting using mathplotlib.

Matplotlib has a great interface working together with pandas.
Therefore I chose to use this package for plotting the graphs.

The commit sets the latest versions still compatible with Python 2.7 and
handles some issues with the dependency chain.
Also some imports are now in use in production and not only in testing
and therefore moved to the production part.

Commit:
Move content creation method to helpers.

I would like to reuse that part of the code and therefore move it into the helper module.

Commit:
Implement graph generator.

This implements a base64 converter and a method to create image tags from matplotlib figures.
The CI Python does not provide thkinter, therefore I use Agg (not e.g. TkAgg) as a backend.

busykoala commented:
I tried to address the changes requested. I hope this is what you meant.

mbaechtold reviewed (approved):
👍
